src/parser/hunter/beastmastery/modules/talents/Stampede.js

```diff
@@ -2,30 +2,143 @@ import React from 'react';
 
 import SPELLS from 'common/SPELLS';
 import SpellLink from 'common/SpellLink';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
 import Analyzer from 'parser/core/Analyzer';
 import ItemDamageDone from 'interface/others/ItemDamageDone';
 import StatisticListBoxItem from 'interface/others/StatisticListBoxItem';
+import StatisticBox from 'interface/others/StatisticBox';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatMilliseconds } from 'common/format';
+
+// The potential amount of hits per target per stampede cast.
+// By checking through various Zek'voz logs, it seems to consistently hit the boss 18 times, except if the boss was moved.
+// By using this number, we can calculate the average amount of targets hit per cast.
+const STAMPEDE_POTENTIAL_HITS = 18;
 
 /**
  * Summon a herd of stampeding animals from the wilds around you that deal damage to your enemies for 12 sec.
  *
  * Example log: https://www.warcraftlogs.com/reports/KQjC6mh74wDBV2Zp#fight=17&type=damage-done&source=21&translate=true
+ * Example log with suggestion triggered: https://www.warcraftlogs.com/reports/z3TH89XagA1hcP7G/#fight=20&source=238
  */
-
 class Stampede extends Analyzer {
+
+  casts = [];
   damage = 0;
+  hits = 0;
+  averageHits = 0;
+  inefficientCasts = 0;
+
 
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.STAMPEDE_TALENT.id);
   }
 
+  get currentCast() {
+    if (this.casts.length === 0) {
+      return null;
+    }
+
+    return this.casts[this.casts.length - 1];
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.STAMPEDE_TALENT.id) {
+      return;
+    }
+
+    this.casts.push({ timestamp: event.timestamp, damage: 0, hits: 0, averageHits: 0 });
+  }
+
   on_byPlayer_damage(event) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.STAMPEDE_DAMAGE.id) {
       return;
     }
-    this.damage += event.amount + (event.absorbed || 0);
+    const damage = event.amount + (event.absorbed || 0);
+
+    this.hits++;
+    this.damage += damage;
+
+    if (this.currentCast !== null) {
+      this.currentCast.hits++;
+      this.currentCast.damage += damage;
+    }
+  }
+
+  on_finished() {
+    this.averageHits = this.hits / this.casts.length / STAMPEDE_POTENTIAL_HITS;
+    this.casts.forEach((cast) => {
+      cast.averageHits = cast.hits / STAMPEDE_POTENTIAL_HITS;
```

Comment:
this *could* be a getter too, but idk how people feel about getters on object literals